### PR TITLE
Observer

### PR DESCRIPTION
**Pull request by jeffreyhelgers**
Implemented the observer design pattern
<organisation

### CODE REVIEW

**Comment by LilideBree** [comment]
Remove unused imports, check all the checkstyles.

<organisation

File: src/main/java/controller/ScreenController.java [hunk 1]
```diff
@@ -91,15 +94,19 @@ public void updateUI() {
...
-        image.relocate(sprite.getX(), sprite.getY());
-        if (sprite.getSpriteChanged()) {
-            image.setImage(new Image(
-            		getClass().getResource(sprite.getImagePath()).toExternalForm()));
-            sprite.setSpriteChanged(false);
-        }
-        image.setRotate(sprite.getR());
+    public void update(SpriteBase sprite) {
+    	int plaats = sprites.indexOf(sprite);
+    	if (plaats >= 0) {
```
Comment (jimhommes) [inline src/main/java/controller/ScreenController.java]: Why is this needed? If this is done because of errors, then the problem should be fixed where the update is called where it shouldn't be.

<bug

File: src/main/java/model/Powerup.java [hunk 2]
```diff
@@ -15,7 +15,8 @@
     private double destx;
     private double desty;
     private boolean ableToPickup;
-    private LevelController levelController;
+    @SuppressWarnings("unused")
+	private LevelController levelController;
```
Comment (jimhommes) [inline src/main/java/model/Powerup.java:19]: If it is unused, then it can be removed. 

<layout

File: src/main/java/model/SpriteBase.java [hunk 3]
```diff
@@ -102,7 +111,10 @@ public void move() {
         x += dx;
         y += dy;
         r += dr;
-
+        
+        observers.get(1).update(this);
```
Comment (jimhommes) [inline src/main/java/model/SpriteBase.java]: Hardcoded, this should be done in a better way.

<bug

File: src/main/java/controller/ScreenController.java (hunk 1 above)
Comment (jeffreyhelgers) [inline src/main/java/controller/ScreenController.java]: This is needed because the ScreenController needs to know if the sprite needs to be added or removed from the screen.

<review

**Comment by LilideBree** [comment]
Checkstyle warnings in Subject and Observer. Subject lines 16 and 23. Observer in 16 and 22.

**Comment by LilideBree** [comment]
:shipit: 

<organisation
